### PR TITLE
Refactor: use feature detection instead of browser sniffing in Env.js

### PR DESCRIPTION
**Pull request by xiongzixiao**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [x] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] It's submitted to the `dev` branch for v2.x (or to a previous version branch), _not_ the `master` branch
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] All tests are passing: https://github.com/vuejs/vue/blob/dev/.github/CONTRIBUTING.md#development-setup
- [ ] New/updated tests are included

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**
Use feature detection instead of browser sniffing，In response to modify the User Agent by user。

### CODE REVIEW

File: src/core/util/env.js [hunk 1]
```diff
@@ -8,12 +8,12 @@ export const inBrowser = typeof window !== 'undefined'
...
 export const UA = inBrowser && window.navigator.userAgent.toLowerCase()
-export const isIE = UA && /msie|trident/.test(UA)
-export const isIE9 = UA && UA.indexOf('msie 9.0') > 0
+export const isIE = inBrowser && document.documentMode
+export const isIE9 = inBrowser && document.documentMode && document.documentMode === 9
 export const isEdge = UA && UA.indexOf('edge/') > 0
 export const isAndroid = (UA && UA.indexOf('android') > 0) || (weexPlatform === 'android')
 export const isIOS = (UA && /iphone|ipad|ipod|ios/.test(UA)) || (weexPlatform === 'ios')
-export const isChrome = UA && /chrome\/\d+/.test(UA) && !isEdge
+export const isChrome = inBrowser && window.chrome
```
Comment (Justineo) [inline src/core/util/env.js]: Opera also supports `window.chrome`.

File: src/core/util/env.js [hunk 2]
```diff
@@ -8,12 +8,12 @@ export const inBrowser = typeof window !== 'undefined'
 export const inWeex = typeof WXEnvironment !== 'undefined' && !!WXEnvironment.platform
 export const weexPlatform = inWeex && WXEnvironment.platform.toLowerCase()
 export const UA = inBrowser && window.navigator.userAgent.toLowerCase()
-export const isIE = UA && /msie|trident/.test(UA)
-export const isIE9 = UA && UA.indexOf('msie 9.0') > 0
+export const isIE = inBrowser && document.documentMode
+export const isIE9 = inBrowser && document.documentMode && document.documentMode === 9
```
Comment (Justineo) [inline src/core/util/env.js]: `inBrowser && document.documentMode === 9` should be enough.

File: src/core/util/env.js (hunk 1 above)
Comment (xiongzixiao) [reply src/core/util/env.js]: ok, fixed it!

File: src/core/util/env.js (hunk 2 above)
Comment (xiongzixiao) [reply src/core/util/env.js]: fixed it, thank you!

File: src/core/util/env.js [hunk 3]
```diff
@@ -8,12 +8,12 @@ export const inBrowser = typeof window !== 'undefined'
...
 export const UA = inBrowser && window.navigator.userAgent.toLowerCase()
-export const isIE = UA && /msie|trident/.test(UA)
-export const isIE9 = UA && UA.indexOf('msie 9.0') > 0
+export const isIE = inBrowser && document.documentMode
+export const isIE9 = inBrowser && document.documentMode === 9
 export const isEdge = UA && UA.indexOf('edge/') > 0
 export const isAndroid = (UA && UA.indexOf('android') > 0) || (weexPlatform === 'android')
 export const isIOS = (UA && /iphone|ipad|ipod|ios/.test(UA)) || (weexPlatform === 'ios')
-export const isChrome = UA && /chrome\/\d+/.test(UA) && !isEdge
+export const isChrome = inBrowser && window.chrome && !window.opr
```
Comment (phanan) [inline src/core/util/env.js]: Have you checked [this StackOverflow's answer](https://stackoverflow.com/a/50942571/794641) (disclaimer: I haven't)?

Comment (xiongzixiao) [reply src/core/util/env.js]: I haven't

**Comment by yyx990803** [comment]
FYI: if something is working and unlikely to change, better not touch it. Refactors like this usually result in potential regressions for no obvious gain.

**Pull request closed without merging**